go/ubirch/ubirch_sim.go

```diff
@@ -282,6 +282,44 @@ func (p *Protocol) GenerateKey(name string, uid uuid.UUID) error {
 	return err
 }
 
+func (p *Protocol) getDataFromTagID(tags []Tag, tagID byte) ([]byte, error) {
+	for _, tag := range tags {
+		if tag.Tag == tagID {
+			return tag.Data, nil
+		}
+	}
+	return nil, errors.New(fmt.Sprintf("did not find tag %x", tagID))
+}
+
+func (p *Protocol) GetUUID(name string) (uuid.UUID, error) {
+	// select SS entry
+	name = "_" + name
+	_, code, err := p.execute(stkAppSsEntrySelect, len(name), hex.EncodeToString([]byte(name)))
+	if err != nil {
+		return uuid.New(), err
```

Comment:
Better to return uuid.nil if this is an error case?

Reply:
done ;)